modules/hal_rpi_pico/pico/config_autogen.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2021, Yonatan Schachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _CONFIG_AUTOGEN_H_
+#define _CONFIG_AUTOGEN_H_
```

Comment:
What's going on with this file name? It looks a little weird and seems to be very close to autoconf.h, which feels inappropriate.

Reply:
Originally `pico-sdk` generates this file, and some `pico-sdk` files include it directly, so that name is essential (it ends up in every picosdk file, which makes it useful for workarounds).

Reply:
Can you please add a source code comment with this information so it doesn't get lost in the PR thread then?